testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -323,17 +396,44 @@ def u():
         return mda.Universe(GRO)
 
     def test_within(self, u):
-        from MDAnalysis.analysis.density import notwithin_coordinates_factory as ncf
-
-        vers1 = ncf(u, 'resname SOL', 'protein', 2, not_within=False, use_kdtree=True)()
-        vers2 = ncf(u, 'resname SOL', 'protein', 2, not_within=False, use_kdtree=False)()
+        vers1 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                      'protein', 2,
+                                                      not_within=False,
+                                                      use_kdtree=True)()
+        vers2 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                      'protein', 2,
+                                                      not_within=False,
+                                                      use_kdtree=False)()
 
         assert_equal(vers1, vers2)
 
     def test_not_within(self, u):
-        from MDAnalysis.analysis.density import notwithin_coordinates_factory as ncf
+        vers1 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                     'protein', 2,
+                                                     not_within=True,
+                                                     use_kdtree=True)()
+        vers2 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                     'protein', 2,
+                                                     not_within=True,
+                                                     use_kdtree=False)()
+        assert_equal(vers1, vers2)
 
-        vers1 = ncf(u, 'resname SOL', 'protein', 2, not_within=True, use_kdtree=True)()
-        vers2 = ncf(u, 'resname SOL', 'protein', 2, not_within=True, use_kdtree=False)()
 
-        assert_equal(vers1, vers2)
+@pytest.mark.parametrize('B', [20, 20.0, 20.123,
+                               np.array([45.678, 67.1, 0.0, 100.2])])
+def test_Bfactor2RMSF(B):
+    values = density.Bfactor2RMSF(B)
+    ref = np.sqrt(3. * B / 8.) / np.pi  # original equation
+    assert_almost_equal(values, ref)
+
+@pytest.mark.parametrize('sigma,ref_shape,ref_gridsum',
+                         [(None, (21, 26, 29), 108.710149),
+                          (1.5,  (21, 26, 29), 160.050167)])
+def test_density_from_PDB(sigma, ref_shape, ref_gridsum):
+    # simple test that the code runs: use B-factors from PDB or fixed sigma
+    D = density.density_from_PDB(PDB_full, delta=2.0, sigma=sigma)
+
+    assert isinstance(D, density.Density)
+    assert_equal(D.grid.shape, ref_shape)
+    assert_almost_equal(D.grid.sum(), ref_gridsum, decimal=6)
+
```

Comment:
Would it be possible to add deprecation warning tests for density_from_PDB and associated classes?